Kiosk/Bid Fulfillment/LoadingViewController.swift

```diff
@@ -126,7 +126,7 @@ public class LoadingViewController: UIViewController {
         let showPlaceHigherButton = placingBid && (!isHighestBidder || reserveNotMet)
         placeHigherBidButton.hidden = !showPlaceHigherButton
 
-        let showAuctionButton = !placingBid || createdNewBidder
+        let showAuctionButton = showPlaceHigherButton || isHighestBidder || (!placingBid && !createdNewBidder)
```

Comment:
This is quite possibly the hairiest part of the app. Another good candidate for MVVM.